mwclient/client.py

```diff
@@ -1264,51 +1286,64 @@ def exturlusage(self, query, prop=None, protocol='http', namespace=None, limit=N
 
         """
 
+        (max_items, api_chunk_size) = handle_limit(limit, max_items, api_chunk_size)
         kwargs = dict(listing.List.generate_kwargs('eu', query=query, prop=prop,
                                                    protocol=protocol,
                                                    namespace=namespace))
-        return listing.List(self, 'exturlusage', 'eu', limit=limit, **kwargs)
+        return listing.List(self, 'exturlusage', 'eu', max_items=max_items,
+                            api_chunk_size=api_chunk_size, **kwargs)
 
     def logevents(self, type=None, prop=None, start=None, end=None,
-                  dir='older', user=None, title=None, limit=None, action=None):
+                  dir='older', user=None, title=None, limit=None, action=None,
+                  max_items=None, api_chunk_size=None):
         """Retrieve logevents as a generator."""
+        (max_items, api_chunk_size) = handle_limit(limit, max_items, api_chunk_size)
         kwargs = dict(listing.List.generate_kwargs('le', prop=prop, type=type,
                                                    start=start, end=end, dir=dir,
                                                    user=user, title=title, action=action))
-        return listing.List(self, 'logevents', 'le', limit=limit, **kwargs)
+        return listing.List(self, 'logevents', 'le', max_items=max_items,
+                            api_chunk_size=api_chunk_size, **kwargs)
 
     def checkuserlog(self, user=None, target=None, limit=10, dir='older',
```

Comment:
I think this `limit=10` default should be moved to to the `api_chunk_size` argument